types/src/vm/environment.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    chain_id::ChainId,
+    on_chain_config::{
+        ConfigurationResource, Features, OnChainConfig, TimedFeatureOverride, TimedFeatures,
+        TimedFeaturesBuilder,
+    },
+    state_store::StateView,
+    vm::configs::{
+        aptos_prod_deserializer_config, aptos_prod_verifier_config, aptos_prod_vm_config,
+        get_paranoid_type_checks, get_timed_feature_override,
+    },
+};
+use move_binary_format::deserializer::DeserializerConfig;
+use move_bytecode_verifier::VerifierConfig;
+use move_vm_runtime::config::VMConfig;
+use std::sync::Arc;
+
+/// A runtime environment which can be used for VM initialization and more.
+#[derive(Clone)]
+pub struct Environment {
+    pub chain_id: ChainId,
+
+    pub features: Features,
+    pub timed_features: TimedFeatures,
+
+    pub deserializer_config: DeserializerConfig,
+    pub verifier_config: VerifierConfig,
+    pub vm_config: VMConfig,
+}
+
+// Wrapper to allow trait bounds on the inner type.
```

Comment:
Hm, is it worse to implement on Environment directly and use Arc of that?

Reply:
Removed the wrapper :)